src/Microsoft.ML.Core/Data/DenseVector.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// A buffer that supports dense vector representation. When an instance of this
+    /// is passed to a row cursor getter, the callee is free to take ownership of
+    /// and re-use the backing data structure (Buffer).
+    /// </summary>
+    public readonly struct DenseVector<T>
```

Comment:
Why do we need this type? Why not use Memory<T> directly? If you need to be able to resize it, could the caller just hold to the original buffer? Or the method filling in data into Memory<T> would return out parameter specifying how many items were filled in.

Reply:
Take a look at https://github.com/dotnet/machinelearning/blob/master/docs/code/VBufferCareFeeding.md#buffer-re-use-as-a-user for the common usage of these vectors.

> Why not use Memory directly?

Because if you only had Memory directly, as soon as you wanted to return a vector that had a length less than the total buffer capacity, now you could no longer get back to use the total capacity of the buffer.

> could the caller just hold to the original buffer?

It's my understanding that the caller doesn't always know how big the original buffer should be, and thus can't create it.

> Or the method filling in data into Memory would return out parameter specifying how many items were filled in.

That won't work with the current `IDataView` `GetGetter` delegate design.

Reply:
What I meant is the following (it's a pattern we use in lots of Span/Memory APIs):
```C#
Memory<byte> buffer = ...
while(true){
    if(TryFillBuffer(buffer, out written)) {
        UseFilledInBuffer(buffer.Slice(0, written));
        break;
    }
    Enlarge(ref buffer);  
} 
```
Could we do something like that?